app/api-v1/routes/attachment/{id}.js

```diff
@@ -1,58 +1,11 @@
 const { getDefaultSecurity } = require('../../../utils/auth')
+const attachmentController = require('../../controllers/Attachment')
 
-// eslint-disable-next-line no-unused-vars
-const { BadRequestError, InternalError, NotAcceptableError } = require('../../../utils/errors')
-const { parseAccept } = require('../../../utils/parseAcceptHeader')
-
-module.exports = function (attachmentService) {
+module.exports = function () {
   const doc = {
-    GET: async function (req, res) {
-      const response = await attachmentService.getAttachmentByID(req.params.id)
-      const filename = response.filename
-      const orderedAccept = parseAccept(req.headers.accept)
-
-      if (filename === 'json') {
-        for (const mimeType of orderedAccept) {
-          if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
-            try {
-              const json = JSON.parse(response.binary_blob)
-              res.status(200).json(json)
-            } catch (error) {
-              throw new InternalError({ message: error })
-            }
-            return
-          }
-          if (mimeType === 'application/octet-stream') {
-            res.status(200)
-            res.set({
-              immutable: true,
-              maxAge: 365 * 24 * 60 * 60 * 1000,
-              'content-disposition': `attachment; filename="${response.filename}"`,
-              'access-control-expose-headers': 'content-disposition',
-              'content-type': 'application/octet-stream',
-            })
-            res.send(response.binary_blob)
-            return
-          }
-        }
-        throw new NotAcceptableError({ message: 'Client file request not supported' })
-      } else {
-        for (const mimeType of orderedAccept) {
-          if (mimeType === 'application/octet-stream' || mimeType === 'application/*' || mimeType === '*/*') {
-            res.status(200)
-            res.set({
-              immutable: true,
-              maxAge: 365 * 24 * 60 * 60 * 1000,
-              'content-disposition': `attachment; filename="${response.filename}"`,
-              'access-control-expose-headers': 'content-disposition',
-              'content-type': 'application/octet-stream',
-            })
-            res.send(response.binary_blob)
-            return
-          }
-          throw new NotAcceptableError({ message: 'Client file request not supported' })
-        }
-      }
+    GET: async (req, res) => {
+      const { status, response } = await attachmentController.get(req)
+      res.status(status).send(response)
```

Comment:
Headers handling?

Reply:
Sorry, I missed that. Was mainly focused on getting this working. @nitro-marky leave it to me I will update now. Thanks for spotting, clearly test coverage is not complete.